system-addon/common/Reducers.jsm

```diff
@@ -91,18 +91,26 @@ function TopSites(prevState = INITIAL_STATE.TopSites, action) {
       }
       return Object.assign({}, prevState, {initialized: true, rows: action.data});
     case at.TOP_SITES_EDIT:
-      return Object.assign({}, prevState, {editForm: {index: action.data.index}});
+      return Object.assign({}, prevState, {editForm: {index: action.data.index}, screenshotRequestFailed: false, screenshotPreview: null});
     case at.TOP_SITES_CANCEL_EDIT:
-      return Object.assign({}, prevState, {editForm: null});
+      return Object.assign({}, prevState, {editForm: null, screenshotRequestFailed: false});
+    case at.SCREENSHOT_FAILED:
+      return Object.assign({}, prevState, {screenshotRequestFailed: true});
```

Comment:
I wonder if some confusion is from a general `SCREENSHOT_FAILED` that is only really used for *preview* screenshots -> `PREVIEW_FAILED`?

Reply:
Renamed this to PREVIEW_FAILED